Chore: (design systems rework) (#343)
chromaui/learnstorybook.com: content/design-systems-for-developers/react/en/test.md

![Visual test components](/design-systems-for-developers/component-visual-testing.gif)

If you’re building a modern UI, visual testing saves your frontend team from time-consuming manual review and prevents expensive UI regressions.

> In the <a href="https://www.learnstorybook.com/design-systems-for-developers/react/en/review/#publish-storybook">previous chapter</a> we learned how to publish Storybook using [Chromatic](https://www.chromatic.com/). We added a bold red border around each `Button` component and then requested feedback from teammates.

Reviewer: ```suggestion
In the <a href="review/#publish-storybook">previous chapter</a> we learned how to publish Storybook using [Chromatic](https://www.chromatic.com/). We added a bold red border around each `Button` component and then requested feedback from teammates.
```

I think we should use relative links?

Reviewer 2: @jonniebigodes mentioned they didn't work quite right in Gatsby for some reason. I can confirm that I get a routing error when I use the relative link.

Reviewer: I'm pretty sure we can make some kind of link that doesn't include a domain work

Author: @tmeasday the reason we went with including the full url for this two situations was due to the fact that Gatsby+ Markdown does not "like" that. When you add a link in markdown relative to another page with a heading with will bubble up to the top of the page. I've tested it and it will always default to the top of the page. 

We could go about this in two ways:

- Add the [gatsby-remark-autolink-headers plugin](https://github.com/gatsbyjs/gatsby/tree/master/packages/gatsby-remark-autolink-headers) to the site and we have a way use the links/navigation better.
- Manually update both `gatsby-ssr` and `gatsby-browser` like it's used in the plugin to include the navigation.

Me personally i'm more inclined to the first one. As at some point it could be extremely beneficial to include a TOC per chapter and proper heading linking, which will be extremely more beneficial for the translations.

A initial approach was put together [here](https://github.com/chromaui/learnstorybook.com/pull/299), per @domyen feedback i thought it over and it would be better to address this properly somewhere in the near future

Author: i've created a issue (#347) for tracking the usage of links in headings here. We can revisit it once we have a little more breathing room. 

Sounds good?

Reviewer: OK!

![Visual test components](/design-systems-for-developers/component-visual-testing.gif)

If you’re building a modern UI, visual testing saves your frontend team from time-consuming manual review and prevents expensive UI regressions.

> In the <a href="/design-systems-for-developers/react/en/review/#publish-storybook"> previous chapter</a> we've learned how to publish our Storybook and get feedback. We'll expand on what we've learned and demo visual testing using Chromatic, an industrial-grade service by the Storybook maintainers.

Reviewer 2: Can we make this URL relative so that we don't have to update it for every framework/translation?

Author: To the best of my knowledge with markdown you cannot do it. With MDX we might achieve this. That's something that would need a bit more of testing.

```

![Result of our first Chromatic build](/design-systems-for-developers/chromatic-first-build.png)
-->

> When the pull request was created, asides from publishing Storybook, Chromatic also captured a set of images for the changes that we made!

Reviewer 2: ```suggestion
When the pull request was created, Chromatic published Storybook then captured a set of images for the changes that we made!
```

Author: Suggested wording applied per feedback. Marking it as resolved.

![List of checks in the pull request](/design-systems-for-developers/chromatic-list-of-checks.png)

Click the "🟡 UI Tests" check to review them.

> ![Second build in Chromatic with changes](/design-systems-for-developers/chromatic-second-build-from-pr.png)

Reviewer 2: iirc the button component has 3 stories. Why are only 2 of them registering changes?

Reviewer 2: Here's the original page for this https://www.chromatic.com/build?appId=5d91884443f1dd00200b8a70&number=2

Yikes! That small tweak resulted in a flood of UI changes.

<!-- ![Second build in Chromatic with changes](/design-systems-for-developers/chromatic-second-build.png) -->

Review them to confirm whether they’re intentional (improvements) or unintentional (bugs).

![Reviewing changes in Chromatic](/design-systems-for-developers/review-changes-pr.png)

If you’re fond of the new font-size, go ahead and accept the changes and commit to git.

> ![Review finished in Chromatic](/design-systems-for-developers/chromatic-success-review.png)

Reviewer 2: This doesn't look the "passed" state in Chromatic.
